differ/windowsdiffer/differ_windows.go

```diff
@@ -0,0 +1,420 @@
+		"UtilityVM/Files/EFI/Microsoft/Boot/BCD.LOG2": "bcd.log2.bak",
+	}
+)
+
+// NewWindowsDiff is a generic implementation of plugin.Differ.
+// NewWindowsDiff is expected to work with any filesystem.
+func NewWindowsDiff(store content.Store) (plugin.Differ, error) {
+	return &windowsDiff{
+		store: store,
+	}, nil
+}
+
+// Apply applies the content associated with the provided digests onto the
+// provided mounts. Archive content will be extracted and decompressed if
+// necessary.
+func (s *windowsDiff) Apply(ctx context.Context, desc ocispec.Descriptor, mounts []mount.Mount) (ocispec.Descriptor, error) {
+	var isCompressed bool
+	switch desc.MediaType {
+	case ocispec.MediaTypeImageLayer, images.MediaTypeDockerSchema2Layer:
+	case ocispec.MediaTypeImageLayerGzip, images.MediaTypeDockerSchema2LayerGzip:
+		isCompressed = true
+	default:
+		// Still apply all generic media types *.tar[.+]gzip and *.tar
+		if strings.HasSuffix(desc.MediaType, ".tar.gzip") || strings.HasSuffix(desc.MediaType, ".tar+gzip") {
+			isCompressed = true
+		} else if !strings.HasSuffix(desc.MediaType, ".tar") {
+			return emptyDesc, errors.Wrapf(errdefs.ErrNotImplemented, "unsupported diff media type: %v", desc.MediaType)
+		}
+	}
+
+	dir, err := ioutil.TempDir("", "extract-")
+	if err != nil {
+		return emptyDesc, errors.Wrap(err, "failed to create temporary directory")
+	}
+	defer os.RemoveAll(dir)
+
+	ra, err := s.store.ReaderAt(ctx, desc.Digest)
+	if err != nil {
+		return emptyDesc, errors.Wrap(err, "failed to get reader from content store")
+	}
+	defer ra.Close()
+
+	r := content.NewReader(ra)
+	if isCompressed {
+		ds, err := compression.DecompressStream(r)
+		if err != nil {
+			return emptyDesc, err
+		}
+		defer ds.Close()
+		r = ds
+	}
+
+	digester := digest.Canonical.Digester()
+	rc := &readCounter{
+		r: io.TeeReader(r, digester.Hash()),
+	}
+
+	layer, parentLayerPaths, err := mountsToLayerAndParents(mounts)
+	if err != nil {
+		return emptyDesc, err
+	}
+
+	size, err := writeLayer(rc, filepath.Dir(mounts[0].Source), layer, parentLayerPaths...)
+	if err != nil {
+		return emptyDesc, errors.Wrap(err, "failed to write layer")
+	}
+
+	// TODO (darrenstahlmsft) use archiver instead of writelayer here. Need parent layer context
+	//if _, err := archive.Apply(ctx, dir, rc); err != nil {
+	//	return emptyDesc, err
+	//}
+
+	// Read any trailing data
+	if _, err := io.Copy(ioutil.Discard, rc); err != nil {
+		return emptyDesc, err
+	}
+
+	return ocispec.Descriptor{
+		MediaType: ocispec.MediaTypeImageLayer,
+		Size:      size,
+		Digest:    digester.Digest(),
+	}, nil
+}
+
+// writeLayer writes a layer from a tar file.
+func writeLayer(layerData io.Reader, home string, layer string, parentLayerPaths ...string) (int64, error) {
+	// TODO (darrenstahlmsft): Make this a re-exec to prevent escalating privilege in main containerd.exe process
+	err := winio.EnableProcessPrivileges([]string{winio.SeBackupPrivilege, winio.SeRestorePrivilege})
```

Comment:
`winio.RunWithPrivileges`?

Reply:
Unfortunately not. This requires the Privileges to be on more than the current thread, as the use of Go threads in the layer writer loses the privileged when thread is swapped. This means the entire process needs the privileges. As a result, this should be done via a second containerd.exe process to just do the layer write. That's next up for changes I'll be making.